packages/web-components/fast-foundation/src/calendar/calendar.ts

```diff
@@ -0,0 +1,339 @@
+import {
+    attr,
+    FASTElement,
+    nullableNumberConverter,
+    observable,
+} from "@microsoft/fast-element";
+import { keyEnter } from "@microsoft/fast-web-utilities";
+import { FoundationElement } from "../foundation-element";
+import type { DayFormat, MonthFormat, WeekdayFormat, YearFormat } from "./date-formatter";
+import { DateFormatter } from "./date-formatter";
+
+/**
+ * Information about a month
+ * @public
+ */
+export type MonthInfo = {
+    month: number;
+    year: number;
+    length: number;
+    start: number;
+};
+
+/**
+ * Calendar information needed for rendering
+ * including the next and previous months
+ * @public
+ */
+export type CalendarInfo = MonthInfo & {
+    previous: MonthInfo;
+    next: MonthInfo;
+};
+
+/**
+ * Caldendar date info
+ * used to represent a date
+ * @public
+ */
+export type CalendarDateInfo = {
+    day: number;
+    month: number;
+    year: number;
+    disabled?: boolean;
+    selected?: boolean;
+};
+
+/**
+ * Calendar component
+ * @public
+ */
+export class Calendar extends FoundationElement {
+    /**
+     * date formatter utitlity for getting localized strings
+     */
+    @observable
+    public dateFormatter: DateFormatter = new DateFormatter();
```

Comment:
Should this be marked as internal?

Reply:
I don't think it needs to be observable either, an internal public prop should do.